src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -343,67 +347,72 @@ def deleteBlocks(self):
-
-        for contName in containerDict:
-            cont = containerDict[contName]
-
-            # Checks if the container is not requested in any sites.
-            # This should never be triggered, but better safe than sorry
-            if not cont['rse']:
-                logging.warning("No rules for container: %s. Its blocks won't be deleted.", contName)
-                continue
+        for blocksSlice in grouper(blockDict, self.sliceSize):
+            logging.info("Handeling %d candidate blocks", len(blocksSlice))
+            # Populate containerDict, assigning each block to its correspondant container
+            for blockName in blocksSlice:
+                container = blockDict[blockName]['dataset']
+                # If the container is not in the dictionary, create a new entry for it
+                if container not in containerDict:
+                    # Set of sites to which the container needs to be transferred
+                    sites = set(x.replace("_MSS", "_Tape") for x in blockDict[blockName]['sites'])
+                    containerDict[container] = {'blocks': [], 'rse': sites}
+                containerDict[container]['blocks'].append(blockName)
+
+            for contName in containerDict:
+                cont = containerDict[contName]
+
+                # Checks if the container is not requested in any sites.
+                # This should never be triggered, but better safe than sorry
+                if not cont['rse']:
+                    logging.warning("No rules for container: %s. Its blocks won't be deleted.", contName)
+                    continue
 
-            try:
-                # Get RSE in which each block is available
-                availableRSEs = self.rucio.getReplicaInfoForBlocks(block=cont['blocks'])
-            except Exception as exc:
-                msg = "Failed to get replica info for blocks in container: %s.\n" % contName
-                msg += "Will retry again in the next cycle. Error: %s" % str(exc)
-                logging.error(msg)
-                continue
+                try:
+                    # Get RSE in which each block is available
+                    availableRSEs = self.rucio.getReplicaInfoForBlocks(block=cont['blocks'], deep=self.deep)
+                except Exception as exc:
+                    msg = "Failed to get replica info for blocks in container: %s.\n" % contName
+                    msg += "Will retry again in the next cycle. Error: %s" % str(exc)
+                    logging.error(msg)
+                    continue
 
-            for blockRSEs in availableRSEs:
-                # If block is available at every RSE its container needs to be transferred, the block can be deleted
-                blockSites = set(blockRSEs['replica'])
-                if cont['rse'].issubset(blockSites):
-                    blocksToDelete.append(blockRSEs['name'])
+                for blockRSEs in availableRSEs:
```

Comment:
hi @germanfgv , Do you think is it possible to check for not all the `availableRSEs`? Just the RSEs that fits the following expression:
`'tier=0|tier=1|T2_CH_CERN'`